packages/dd-trace/test/appsec/sdk/index.spec.js

```diff
@@ -0,0 +1,342 @@
+const proxyquire = require('proxyquire')
+const agent = require('../../plugins/agent')
+const tracer = require('../../../../../index')
+const appsec = require('../../../src/appsec')
+const Config = require('../../../src/config')
+const getPort = require('get-port')
+const axios = require('axios')
+const RuleManager = require('../../../src/appsec/rule_manager')
+const fs = require('fs')
+const crypto = require('crypto')
+
+describe('Appsec SDK', () => {
+  describe('Check internal callings', () => {
+    const tracer = {}
+    const mockReq = { protocol: 'https' }
+    const mockRes = { headersSent: false }
+    const isUserBlocked = sinon.stub()
+    const storage = { getStore: () => {
+      return { req: mockReq, res: mockRes }
+    } }
+    const block = sinon.stub()
+    const root = sinon.stub().returns({})
+    const loadTemplates = sinon.stub()
+    const mockSetTag = sinon.stub()
+    const mockRootSpan = {
+      context: () => {
+        return { _tags: { 'usr.id': 'mockUser' } }
+      },
+      setTag: mockSetTag }
+    const getRootSpan = sinon.stub().returns(mockRootSpan)
+
+    const AppsecSDK = proxyquire('../../../src/appsec/sdk', {
+      './user_blocking': { isUserBlocked },
+      '../blocking': { block, loadTemplates },
+      '../../plugins/util/web': { root },
+      '../../../../datadog-core': { storage },
+      './utils': { getRootSpan }
+    })
+
+    const sdk = new AppsecSDK(tracer)
+
+    beforeEach(() => {
+      isUserBlocked.reset()
```

Comment:
If you initialize all the methods here, you can prevent doing this `reset` here, these `reset()` are easy to forget